practices/yagni_principle.md

```diff
@@ -0,0 +1,90 @@
+---
+slug: yagni-principle
+
+stages: 
+  - development
+
+short_description: YAGNI ('You aren't gonna need it') principle prevents a programmer from adding unnecessary features to the code.
+
+tags: 
+  - complexity
+  - code
+  - quality
+  - cost
+  - features
+  - unnecessary
+
+
+keywords: 
+  - YAGNI
+  - extreme
+  - programming
+  - principle
+  - XP
+  - complexity
+
+covered_by_dx_scanner: false 
+
+---
+
+
+# YAGNI Principle
+
+**TL;DR**
+
+YAGNI stands for 'You aren't gonna need it'.This principle states that only the required functionalities should be implemented.
+
+
+
+## What Is YAGNI Principle
+
+YAGNI is an extreme programming principle (XP) which states that you must not implement features that you think may be needed in the future. Only those functionalities should be implemented which are needed as per current requirements. 
+
+![YAGNI Principle](/files/yagni_principle.png) 
+[YAGNI Principle](http://www.agilenutshell.com/yagni)
+
+
+
+## Why You Might Want YAGNI Principle
+
+This principle helps you to focus on the required features and not add unnecessary functionality. As a programmer, you sometimes try to foresee the future of functionalities. You presume that additional requirements might come up in the future. There is an assumption that the present code should encapsulate this.
+A lot of effort and time is involved in implementing new functionalities. Implementing additional functionalities can lead to schedule slippage, code complexity apart from unnecessary code. These features need to be maintained and may even need to be modified in the future. There are chances that the functionalities may become obsolete and may never be needed.
+
+
+
+## Problems YAGNI Principle Helps to Solve
+
+- [Increased Cost](/problems/increased-cost)
+- [Meaningless Work](/problems/meaningless-work)
+- [Poor code quality](/problems/poor-code-quality)
+- [Unnecessary Features](/problems/unnecessary-features)
+
+
+
+## How to Implement YAGNI Principle
+
+**Clarity on requirements and deadlines**
+
+You must have a clear picture of the requirements and targets to prevent deviations from the required design.
+
+**Analysis**
+
+Think if you will need the extra features sometime soon or are you assuming they may be a part of the distant future.
```

Comment:
```suggestion
Ask yourself whether you will need these features  soon or if you are just assuming they might be important.```((

Suggestion for clarity's sake, to show that the important distinction is in "will you need them or are you assuming that you will need them". In the original, a non-native speaker can instead think that the distinction is in the timing (soon vs distant future).